src/app/games/rectanglz/tests/rectanglz.component.spec.ts

```diff
@@ -0,0 +1,187 @@
+        }));
+
+        it('should cancel move attempt when clicking on invalid landing', fakeAsync(async() => {
+            // Given a board with a selected piece
+            await testUtils.expectClickSuccess('#click_0_0');
+            testUtils.expectElementToHaveClasses('#piece_0_0', ['base', 'player0-fill', 'selected-stroke']);
+
+            // When licking on invalid landing space
+            await testUtils.expectClickFailure('#click_5_5', RectanglzFailure.MAX_DISTANCE_IS_(2));
+
+            // Then the piece should no longer be selected
+            testUtils.expectElementToHaveClasses('#piece_0_0', ['base', 'player0-fill']);
+        }));
+
+        it('should change selected piece when clicking on another of your piece', fakeAsync(async() => {
+            // Given a board with a selected piece
+            await testUtils.expectClickSuccess('#click_0_0');
+            testUtils.expectElementToHaveClass('#piece_0_0', 'selected-stroke');
+
+            // When clicking on another piece
+            await testUtils.expectClickSuccess('#click_7_7');
+
+            // Then new clicked piece should be highlighted, not the previous
+            testUtils.expectElementNotToHaveClass('#piece_0_0', 'selected-stroke');
+            testUtils.expectElementToHaveClass('#piece_7_7', 'selected-stroke');
+        }));
+
+        it('should cancel move without throwing when clicking on piece again', fakeAsync(async() => {
+            // Given the initial board with one selected piece
+            await testUtils.expectClickSuccess('#click_0_0');
+            testUtils.expectElementToHaveClasses('#piece_0_0', ['base', 'player0-fill', 'selected-stroke']);
+
+            // When clicking that piece again
+            await testUtils.expectClickSuccess('#click_0_0');
+
+            // Then the piece should no longer be selected
+            testUtils.expectElementToHaveClasses('#piece_0_0', ['base', 'player0-fill']);
+        }));
+
+        it('should show captured piece after move', fakeAsync(async() => {
+            // Given a board with a selected piece
+            const state: RectanglzState = new RectanglzState([
+                [_, _, _, _, _, _, _, X],
+                [_, _, _, _, _, _, _, _],
+                [O, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _],
+                [X, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, O],
+            ], 2);
+            await testUtils.setupState(state);
+            await testUtils.expectClickSuccess('#click_0_2');
+
+            // When clicking on valid landing space
+            const move: RectanglzMove = RectanglzMove.from(new Coord(0, 2), new Coord(0, 4)).get();
+            await testUtils.expectMoveSuccess('#click_0_4', move);
+
+            // Then captured piece should be displayed
```

Comment:
relire tous les GWT de ce fichier et ajouter les articles plz! "Then the captured piece*"